_source/_posts/2023-04-19-api-integrations.md

```diff
@@ -0,0 +1,112 @@
+
+But, if you're planning on creating an Okta integration publicly available on the [Okta Integration Network (OIN)](https://developer.okta.com/docs/guides/okta-integration-network/), you'll want to a better mechanism for making your integration application more discoverable to potential customers. When you have a public integration, you delegate scope configuration to the consumer. There's a temptation to either allow too much access or, conversely, limit access, causing frustrations that your integration needs fixes to work as intended.  
+
+**When to use OAuth authorization for integrations**
+
+You should use OAuth as the preferred authorization mechanism. Use OAuth when you are creating an internal integration process.
+
+## API Integration Services
+
+API Integration Services have all of the advantages of OAuth authorization, with the added ability for discoverability for public integrations! You'll use the standard OAuth Client Credentials flow, which means you can use libraries that support it for your framework/language. Plus, you'll define the scopes you need to ensure you have the ones required for running your integration and provide transparent assurance to consumers on the permissions you require and those you don't require. When they understand the access levels, consumers may feel comfortable adding integrations to their organization.
+
+You'll create the machine-to-machine process, test it locally, and then submit the integration to the OIN for public consumption. Okta customer administrators will see all available API Service Integrations in the Okta Admin Console. They can find the integration they want to add, review the requested scopes, and authorize it. They receive a Client ID and Client Secret that they provide to you, the integration provider, along with the Okta Domain so you can make the calls to retrieve the access token.
+
+**Considerations with using API Integration Services**
+
+You'll use standard OAuth flows, so you'll have a secure authorization mechanism, but it is a 2-step process to get the access token. You'll also need to define the scopes you need beforehand, which will help customers feel more comfortable using your integration.
+
+**When to use API Integration Services**
+
+Use API Integration Services when you create machine-to-machine integration for general use within the OIN. You'll have security, discoverability, and clarity for the users when you use this approach.
+
+## Decide which API authorization mechanism to use
+
+When calling Okta APIs, we recommend using API Integration Services first, then OAuth, and finally API Tokens as shown in the following flowchart:
```

Comment:
@vanngo-okta Since this was written as a POV of an ISV writing an integration, suggesting the following tweak on your suggestion. LMK what you think

> When calling Okta APIs, we recommend using OAuth 2.0. If you're creating a public service app accessible through the OIN, use API Service Integration. If you have limitations using OAuth 2.0 for your use case, then use API Tokens as shown in the following flowchart:

Reply:
I like this update 👍 . And would probably go further about OAuth in the API Service Integration:

> When calling Okta APIs, we recommend using OAuth 2.0. If you're creating a public service app accessible through the OIN, use OAuth 2.0 with the API Service Integration. If you have limitations using OAuth 2.0 for your use case, then use API Tokens as shown in the following flowchart: